src/Bicep.LangServer/Handlers/BicepDeployCommandHandler.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager;
+using Bicep.Core.Configuration;
+using Bicep.Core.Registry.Auth;
+using Bicep.LanguageServer.Deploy;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    public class BicepDeployCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string, string, string, string, string, string>
+    {
+        private readonly ITokenCredentialFactory credentialFactory;
+        private readonly IConfigurationManager configurationManager;
+        private readonly IDeploymentCollectionProvider deploymentCollectionProvider;
+
+        public BicepDeployCommandHandler(
+            ISerializer serializer,
+            IConfigurationManager configurationManager,
+            ITokenCredentialFactory credentialFactory,
+            IDeploymentCollectionProvider deploymentCollectionProvider)
+            : base(LangServerConstants.Deploy, serializer)
+        {
+            this.credentialFactory = credentialFactory;
+            this.configurationManager = configurationManager;
+            this.deploymentCollectionProvider = deploymentCollectionProvider;
+        }
+
+        public override async Task<string> Handle(
```

Comment:
Would be much easier to understand if you implemented a record for parameters, see for instance BicepDeploymentGraphHandler.

Reply:
Sure, updated here:
https://github.com/Azure/bicep/commit/830f849d8bee5696579b2c0596095fd6828924b0

Thanks!